### ISSUE TITLE
Commit #1c8a7a39c9 introduced a dependency on cloudformation:DescribeStacksCommand

### ISSUE THREAD

**Issue by berenddeboer** [serverless-stack/sst#3006]
But the function was never given permission to do this.

CloudWatch log error:

```
2023-06-20T23:04:03.857Z	9c08304a-231f-42e2-bedf-367c9884363e	INFO	[provider-framework] AccessDenied: User: arn:aws:sts::257611358457:assumed-role/sst-rds-serverless-v1-clu-DatabaseMigrationHandler-1LLXG8DHG1P3R/sst-rds-serverless-v1-clu-DatabaseMigrationHandler-u6MhORefMNmF is not authorized to perform: cloudformation:DescribeStacks on resource: arn:aws:cloudformation:ap-southeast-2:257611358457:stack/sst-rds-serverless-v1-cluster-dev/026d1450-0fbe-11ee-94a2-0a435c2a0240 because no identity-based policy allows the cloudformation:DescribeStacks action
    at throwDefaultError (file:///var/task/index.mjs:1812:24)
    at deserializeAws_queryDescribeStacksCommandError (file:///var/task/index.mjs:36226:45)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async file:///var/task/index.mjs:818:24
    at async file:///var/task/index.mjs:17201:22
    at async file:///var/task/index.mjs:16079:42
    at async file:///var/task/index.mjs:14895:24
    at async isStackRollingBack (file:///var/task/index.mjs:49039:16)
    at async file:///var/task/index.mjs:49023:18
    at async file:///var/task/index.mjs:48981:7 {
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 403,
    requestId: '586c9a5f-b3ed-4a55-b292-aa14ceff778d',
    extendedRequestId: undefined,
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  },
  Type: 'Sender',
  Code: 'AccessDenied'
}

```

**Comment by jayair** [comment]
Let's put this on #help? Thanks.

**Comment by berenddeboer** [comment]
Workaround, inside your stack add this:

```ts
import * as lambda from "aws-cdk-lib/aws-lambda"
import * as iam from "aws-cdk-lib/aws-iam"

  const cluster = new RDS(stack, "Database", {
    ...
  })
  const migration_handler = cluster.node.tryFindChild("MigrationHandler") as lambda.Function
  if (migration_handler) {
    migration_handler?.addToRolePolicy(new iam.PolicyStatement({
      sid: "AllowAccessToOwnStack",
      effect: iam.Effect.ALLOW,
      actions: ["cloudformation:DescribeStacks"],
      resources: [`arn:aws:cloudformation:${stack.region}:${stack.account}:stack/${stack.stackName}/*`],
    }))
  }
```

**Comment by berenddeboer** [comment]
This got just fixed in aa36e945ee.
